scheduler/spread.go

```diff
@@ -133,6 +169,48 @@ func (iter *SpreadIterator) Next() *RankedNode {
 	}
 }
 
+// evenSpreadScoreBoost is a scoring helper that calculates the score
+// for the option when even spread is desired (all attribute values get equal preference)
+func evenSpreadScoreBoost(pset *propertySet, option *structs.Node) float64 {
+	combinedUseMap := pset.GetCombinedUseMap()
+	if len(combinedUseMap) == 0 {
+		// Nothing placed yet, so return 0 as the score
+		return 0.0
+	}
+	// Get the nodes property value
+	nValue, ok := getProperty(option, pset.targetAttribute)
+	currentAttributeCount := uint64(0)
+	if ok {
```

Comment:
If `!ok` shouldn't we return a negative score?

Reply:
good catch, will fix. 
i conflated !ok with the case where that specific value is not present in the combinedUse map.